src/main/java/spireMapOverhaul/zones/gremlinTown/patches/HordePatches.java

```diff
@@ -0,0 +1,200 @@
+            private Locator() {}
+
+            @Override
+            public int[] Locate(CtBehavior behavior) throws Exception {
+                Matcher matcher = new Matcher.FieldAccessMatcher(GameActionManager.class, "totalDiscardedThisTurn");
+                return LineFinder.findInOrder(behavior, matcher);
+            }
+        }
+    }
+
+    @SpirePatch2(
+            clz = AbstractRoom.class,
+            method = "render"
+    )
+    public static class renderPatch {
+        @SpireInsertPatch (
+                locator = Locator.class,
+                localvars = {"sb"}
+        )
+        public static void Insert(SpriteBatch sb) {
+            if (AbstractDungeon.lastCombatMetricKey.equals(GremlinTown.GREMLIN_HORDE))
+                HordeHelper.render(sb);
+        }
+
+        public static class Locator extends SpireInsertLocator {
+            private Locator() {}
+
+            @Override
+            public int[] Locate(CtBehavior behavior) throws Exception {
+                Matcher matcher = new Matcher.MethodCallMatcher(MonsterGroup.class, "render");
+                return LineFinder.findInOrder(behavior, matcher);
+            }
+        }
+    }
+
+    @SpirePatch2(
+            clz = AbstractPlayer.class,
+            method = "onVictory"
+    )
+    public static class victoryPatch {
+        @SpirePrefixPatch
+        public static void Prefix() {
+            if (AbstractDungeon.lastCombatMetricKey.equals(GremlinTown.GREMLIN_HORDE))
+                HordeHelper.onVictory();
+        }
+    }
+
+    @SpirePatch2(
+            clz = AbstractMonster.class,
+            method = "die",
+            paramtypez = boolean.class
+    )
+    public static class backAttackPatch {
+        @SpirePostfixPatch
+        public static void Postfix() {
+            if (AbstractDungeon.lastCombatMetricKey.equals(GremlinTown.GREMLIN_HORDE)) {
+                HordeHelper.calculateBackAttack();
+            }
+        }
+    }
+
+    @SpirePatch2(
+            clz = GainBlockRandomMonsterAction.class,
+            method = "update"
+    )
+    public static class shieldPatch {
+        @SpireInsertPatch (
+                locator = Locator.class,
+                localvars = {"validMonsters"}
+        )
+        public static void Insert(ArrayList<AbstractMonster> validMonsters) {
+            validMonsters.removeIf(AbstractCreature::isDeadOrEscaped);
+        }
+
+        public static class Locator extends SpireInsertLocator {
+            private Locator() {}
+
+            @Override
+            public int[] Locate(CtBehavior behavior) throws Exception {
+                Matcher matcher = new Matcher.MethodCallMatcher(ArrayList.class, "isEmpty");
+                return LineFinder.findInOrder(behavior, matcher);
+            }
+        }
+    }
+
+    @SpirePatch2(
+            clz = ApplyPowerAction.class,
+            method = "update"
+    )
+    public static class stackingBackAttackPatch {
```

Comment:
Wouldn't it have been easier to make your own back attack power?

Reply:
No, I'd still have to patch it regardless, it isn't something you can set in your own power.